### DISCUSSION TITLE
#include <Radiolib/SX1262.h> is available to use less space than the hole library?

### DISCUSSION THREAD

**Discussion by richonguzman** [jgromes/RadioLib#1098] [Q&A]
I'm running out of space in the ESP32 board and wanted to ask (as I could not find any valid answer about it) if the library could be imported/included by only a part of it.

in python you can:

"from radiolib import sx1262" or something like that...

can you do :

#include <Radiolib/SX1262.h>

or something like that to avoid importing the hole library ?

**Comment by jgromes** [comment]
> or something like that to avoid importing the hole library ?

That's not how a compiled language works. All the unused classes will not be present in the final library. If a symbol is not being used by anything, it will not be linked and hence it will not be in your output file. Feel free to explore the compiled ELF file with tools like objdump.

Also, not really an issue - converting to a discussion instead.

**Reply by richonguzman** [reply]
thanks for the info, this is clearly why I did not find a valid answer before as there is no answer to it in the way I was looking for (compared to python and such)

So as you say, the compiled library takes what it needs and all other clases are "omitted" in .bin files

thanks a lot again for this ;)

**Comment by StevenCellist** [comment]
To follow up on this: while the unused classes get compiled out, the PhysicalLayer is _not_ and results in roughly 6kB of 'bloat'. Running LoRaWAN through the fully-fledged RadioLib uses about 40kB flash, but the compiler is unable to optimize a large number of unused functions away due to the inheritance through PhysicalLayer. Bypassing the PhysicalLayer reduces the compiled size from 40kB to about 34kB.
